src/nms_adpcm.c

```diff
@@ -258,7 +260,12 @@ nms_adpcm_update (struct nms_adpcm_state *s)
 		} ;
 
 	/* Compute the signal estimate. */
-	s->s_e = s->a [0] * s->s_r [0] + s->a [1] * s->s_r [1] + s->s_ez ;
+	signal_estimate = (int64_t) s->a [0] * s->s_r [0] + (int64_t) s->a [1] * s->s_r [1] + s->s_ez ;
+	if (signal_estimate > INT_MAX)
+		signal_estimate = INT_MAX ;
+	else if (signal_estimate < -INT_MAX)
+		signal_estimate = -INT_MAX ;
+	s->s_e = (int) signal_estimate ;
```

Comment:
@ codeowners
I'm making a bold assertion that it is logically correct to clamp the signal estimate value here, rather than erroring. Let me know if that seems incorrect.

Reply:
I'm going to push back on this one a little. I think this is a false-positive.

The signal estimate going out of bounds indicates that bad input data, so the result that comes out being undefined is ok. In essence, any fix here would be "you put garbage in, what kind of garbage out would you like?"

Critical to this is that NMS ADPCM has no variable control flow: there are no branches. The same functions are called always in the same order, all loops are constant (could be unrolled), and all array access is bounded. Static analysis could inline all functions and turn `nms_adpcm_decode_sample()` and `nms_adpcm_encode_sample()` each into one non-branching function. That they are functions is because the same code is used but in reverse order for encode and decode.

Is there a way to tell the fuzzer to ignore this? Or a choice of datatypes that makes this more explicit?

The choice of `int` as a datatype for the state variables was one of efficiency. "If you don't know what to pick, pick `int`". The only requirement is that they be signed and at least 16 bit. `int_fast16_t` would be a more explicit choice.

The signal estimate and signal zero estimate calculation is actually 16-bit decimal multiplication in fixed-point 32 bits.

Reply:
That's fair. If you would really prefer to ignore the error, I can investigate if that is possible. Patch V2 just cast the signal components to satisfy the fuzzer. Let me know what you want to do.

Reply:
Have a look at cb65f952ad1d5be7d4ba41d6c3f1c3ff9e0916b2 :)

Reply:
Yep. That looks good to me. I'll replace my implementation with that in V3 (if only to keep the CVE commits relatively clean.) Thanks for the expertise!